opencga-analysis/src/main/java/org/opencb/opencga/analysis/alignment/qc/AlignmentQcAnalysis.java

```diff
@@ -70,212 +86,291 @@ protected void check() throws Exception {
+            throw new ToolException("Error creating SAMtools flagstat output folder: " + outPath, e);
+        }
+
+        // Prepare parameters
+        SamtoolsWrapperParams samtoolsWrapperParams = new SamtoolsWrapperParams("flagstat", catalogBamFile.getId(), null,
+                new HashMap<>());
+
+        // Execute the Samtools flag stats analysis and add its step attributes if exist
+        ExecutionResult executionResult = toolRunner.execute(SamtoolsWrapperAnalysis.class, study, samtoolsWrapperParams, outPath,
+                null, token);
+        addStepAttributes(executionResult);
 
+        // Check execution status
+        if (executionResult.getStatus().getName() != Status.Type.DONE) {
+            throw new ToolException("Something wrong happened running the Samtools flagstat analysis. Execution status = "
+                    + executionResult.getStatus().getName());
+        }
+
+        // Check results and update QC file
+        Path flagStatsFile = AlignmentFlagStatsAnalysis.getResultPath(outPath.toAbsolutePath().toString(), catalogBamFile.getName());
+        java.io.File stdoutFile = outPath.resolve(DockerWrapperAnalysisExecutor.STDOUT_FILENAME).toFile();
+        List<String> lines ;
+        try {
+            lines = readLines(stdoutFile, Charset.defaultCharset());
+        } catch (IOException e) {
+            throw new ToolException("Error reading running Samtools flagstat results", e);
+        }
+        if (CollectionUtils.isNotEmpty(lines) && lines.get(0).contains("QC-passed")) {
             try {
-                if (runFastqc) {
-                    // FastQC metrics
-                    params = new AlignmentFastQcMetricsParams(analysisParams.getBamFile(), null)
-                            .toParams(new ObjectMap(ParamConstants.STUDY_PARAM, study));
-
-                    OpenCGAResult<Job> fastQcMetricsJobResult = catalogManager.getJobManager()
-                            .submit(study, AlignmentFastQcMetricsAnalysis.ID, Enums.Priority.MEDIUM, params, null,
-                                    "Job generated by " + getId() + " - " + getJobId(), Collections.emptyList(), Collections.emptyList(),
-                                    token);
-                    fastQcMetricsJobId = fastQcMetricsJobResult.first().getId();
-                    addEvent(Event.Type.INFO, "Submit job " + fastQcMetricsJobId + " to compute FastQC metrics ("
-                            + AlignmentFastQcMetricsAnalysis.ID + ")");
-                }
-            } catch (CatalogException e) {
-                addWarning("Error launching job for Alignment FastQC Metrics Analysis: " + e.getMessage());
+                FileUtils.copyFile(stdoutFile, flagStatsFile.toFile());
+            } catch (IOException e) {
+                throw new ToolException("Error copying Samtools flagstat results", e);
             }
+        } else {
+            throw new ToolException("Something wrong happened running Samtools flagstat analysis");
```

Comment:
Your future self will appreciate if you add some extra information here. Like the number of lines, and/or the actual first line